zebedee-cms/src/main/java/com/github/onsdigital/zebedee/session/service/NewSessionsServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+    }
+
+    /**
+     * Create a new {@link Session} for the user.
+     *
+     * @param user the {@link User} the session is being created for.
+     * @return a {@link Session} instance for the user.
+     * @throws IOException problem creating a session.
+     */
+    @Override
+    public Session create(User user) throws IOException {
+        SessionCreated sessionCreated = client.createNewSession(user.getEmail());
+        com.session.service.Session clientSession = client.getSessionByID(sessionCreated.getId());
+
+        if (clientSession == null) {
+            throw new IOException();
+        }
+
+        Session session = createZebedeeSession(clientSession);
+        return session;
+    }
+
+    /**
+     * Get a {@link Session} from the provided {@link HttpServletRequest}.
+     *
+     * @param request the {@link HttpServletRequest} to get the session for.
+     * @return a {@link Session} instance if once exists returns null otherwise.
+     * @throws IOException for any problem getting a sesison from the request.
+     */
+    @Override
+    public Session get(HttpServletRequest request) throws IOException {
+        String token = RequestUtils.getSessionId(request);
+        return get(token);
+    }
+
+    /**
+     * Get a {@link Session} by it's ID.
+     *
+     * @param id the ID of the session to get,
+     * @return the {@link Session} instance if it exists and is not expired.
+     * @throws IOException for any problems getting the session.
+     */
+    @Override
+    public Session get(String id) throws IOException {
+        com.session.service.Session clientSession = client.getSessionByID(id);
+
+        Session session = createZebedeeSession(clientSession);
+        return session;
+    }
+
+    /**
+     * Find a {@link Session} assocaited with the user email.
+     *
+     * @param email the user email address to find.
+     * @return a {@link Session} for the requested email if it exists and is not expired. Return null otherwise.
+     * @throws IOException for any problems getting the session.
+     */
+    @Override
+    public Session find(String email) throws IOException {
+        com.session.service.Session clientSession = client.getSessionByEmail(email);
+
+        Session session = createZebedeeSession(clientSession);
+        return session;
+    }
+
+    /**
+     * Check if an active {@link Session} exists with the provided ID.
+     *
+     * @param id the session ID to find.
+     * @return true if an active session exists with this ID, false otherwise.
+     * @throws IOException problem checking the session exists.
+     */
+    @Override
+    public boolean exists(String id) throws IOException {
+        return false;
+    }
+
+    /**
+     * Check if the provided {@link Session} is expired.
+     *
+     * @param session the {@link Session} to check.
+     * @return true if expired, false otherwise.
+     */
+    @Override
+    public boolean expired(Session session) {
+        return false;
```

Comment:
I guess the expiry will be implemented in the future, is it correct?

Reply:
Expiry is all handled in the sessions api, I'm pretty sure I can remove this.

Reply:
Am I right in saying this is a legacy method that is no longer needed? If so update it to always throw an `UnsupportedOperationException` just makes it clear and obvious its being called when it shouldn't be. Once we've migrated completely to the new sessions service we can consider deleting the method from the interface, technical that is naughty but its only used within Zebedee so it's unlikely to affect anyone but us.

Reply:
Makes sense, have updated it